api/v1/processingjob/processingjob_types.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This annotation signals to the generator that there are types in this file
+// that need DeepCopy methods.
+// +kubebuilder:object:generate=true
+
+// This annotation signals that the types in this file should be a CRD in the group.
+// +groupName=sagemaker.aws.amazon.com
+
+package v1
+
+import (
+	commonv1 "github.com/aws/amazon-sagemaker-operator-for-k8s/api/v1/common"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// ProcessingJobSpec defines the desired state of ProcessingJob
+type ProcessingJobSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	AppSpecification *commonv1.AppSpecification `json:"appSpecification"`
+
+	// +kubebuilder:validation:MaxItems=100
+	Environment []*commonv1.KeyValuePair `json:"environment,omitempty"`
+
+	NetworkConfig *commonv1.ProcessingNetworkConfig `json:"networkConfig,omitempty"`
+
+	// +kubebuilder:validation:MaxItems=10
+	ProcessingInputs []*commonv1.ProcessingInput `json:"processingInputs,omitempty"`
+
+	ProcessingOutputConfig *commonv1.ProcessingOutputConfig `json:"processingOutputConfig,omitempty"`
+
+	ProcessingResources *commonv1.ProcessingResources `json:"processingResources"`
+
+	// +kubebuilder:validation:MinLength=20
+	// +kubebuilder:validation:MaxLength=2048
+	// +kubebuilderrrr:validation:Pattern="^arn:aws[a-z\-]*:iam::\d{12}:role/?[a-zA-Z_0-9+=,.@\-_/]+$"
```

Comment:
`kubebuilderrrr` -> `kubebuilder`
Also do we enforce this pattern for any other ARN? This could also be caught on the server side.

Reply:
client side validation is better ? i am fine either ways